dbms/src/Storages/KVStore/Decode/RegionTable.cpp

```diff
@@ -439,12 +445,12 @@ void RegionTable::replaceRegion(const RegionPtr & old_region, const RegionPtr &
         // `old_region` will no longer contribute to the memory of the table.
         auto keyspace_id = region_range_keys->getKeyspaceID();
         auto table_id = region_range_keys->getMappedTableID();
-        auto & table = getOrCreateTable(keyspace_id, table_id);
-        old_region->resetRegionTableCtx();
+        getOrCreateTable(keyspace_id, table_id);
+        old_region->clearAllData();
```

Comment:
`RegionTable` should not manipulate `Region`'s data. It only manages mapping from Region to Table. It is `KVStore` and Proxy's job to manipulate data.